crates/bitwarden-core/src/mobile/crypto.rs

```diff
@@ -334,7 +334,7 @@ pub(super) fn derive_pin_key(
 
     Ok(DerivePinKeyResponse {
         pin_protected_user_key,
-        encrypted_pin: pin.encrypt_with_key(user_key)?,
+        encrypted_pin: pin.encrypt_with_key(user_key, ContentFormat::Utf8)?,
```

Comment:
I am slightly concerned with `ContentFormat` leaking outside the crypto crate. Having to rely on the callers to set the content seems potentially fragile.

I wonder if we could enforce compile type safety here, a loose thought I have is what if we define

```rust

struct EncryptedString {}

impl Encryptable for EncryptedString {
  fn content_type() {
    ContentFormat::Utf8
  }
}
```

Or in this case 

```rust
struct SymmetricCryptoKey {}

impl Encryptable for SymmetricCryptoKey {
  fn content_type() {
    ContentFormat::Utf8
  }
}
```

That said this is a key right, so `Utf8` is incorrect and it should be Octat.

Reply:
I think @dani-garcia explored something tangented but we had trouble getting a decent non clunky interface for it.